release-calendar/test/repository-service.ts

```diff
@@ -39,9 +39,11 @@ export class RepositoryService {
     return this.releaseRepository.find({relations: ['promotions']});
   }
 
-  async createRelease(release: Release): Promise<Release> {
+  async createRelease(release: Release, date?: Date): Promise<Release> {
     const entity = await this.releaseRepository.save(release);
-    const promotion = new Promotion(entity, Channel.CREATED, Channel.NIGHTLY);
+    const promotion = date
+      ? new Promotion(entity, Channel.CREATED, Channel.NIGHTLY, date)
```

Comment:
What happens if you pass `undefined`/`null` as the 4th argument? If it behaves the same as not passing it, you can skip the `?:` and just create the new Promotion instance passing in `date`

Reply:
You're right :) i cleaned it up